### DISCUSSION TITLE
Problem with retry in case of PM2 restart

### DISCUSSION THREAD

**Discussion by srivatsan-xoxoday** [bbc/sqs-consumer#432] [Help] [answered]
### Summary

We are using node processes running on PM2 to consume the messages from SQS via sqs-consumer library. When our code throws a "Error", the SQS library is properly retrying the messages. But in case there is a PM2 restart of a VM restart/shutdown, the message is completely deleted. We have tried configuring Dead Letter Queue and in those cases of PM2 restart/shutdown the unprocessed messages do not appear even in the DLQ. 

We need help with solving this as this is resulting in major failures in processing the events.

**Comment by nicholasgriffintn** [comment, marked as answer]
Sorry, not sure we can really do anything here, we can't really help with your app, happy to leave this discussion up in case anyone has any advice though.
